tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ImportLiferayModuleProjectOpMethods.java

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core.modules;
+
+import com.liferay.ide.core.ILiferayProjectImporter;
+import com.liferay.ide.core.LiferayCore;
+import com.liferay.ide.project.core.ProjectCore;
+import com.liferay.ide.project.core.util.ProjectUtil;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.sapphire.modeling.ProgressMonitor;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.ProgressMonitorBridge;
+
+/**
+ * @author Andy Wu
+ */
+public class ImportLiferayModuleProjectOpMethods
+{
+
+    public static final Status execute( final ImportLiferayModuleProjectOp op, final ProgressMonitor pm )
+    {
+        final IProgressMonitor monitor = ProgressMonitorBridge.create( pm );
+
+        monitor.beginTask( "Importing Module project...", 100 );
+
+        Status retval = null;
+
+        try
+        {
+            String location = op.getLocation().content().toOSString();
+            String buildType = getBuildType( location );
+            ILiferayProjectImporter[] importers = LiferayCore.getImporter( buildType );
```

Comment:
We should have a method that just gets the imports and then asks "canImport(IPath location)" and only call import on that one.